### PR TITLE
reorganize repo and add geo_concerns

### PR DESCRIPTION
**Pull request by eliotjordan**
#### What's going on here?

1. Includes rebased commits from @yulgit1 and @drh-stanford to preserve contribution history.
2. Adds geo_concerns application for developing Hydra Geo models.
2. Reorganization of repo.

I became really frustrated while attempting to flesh out a pcdm-geo-models gem that mirrors the structure of curation_concerns-models. There was simply too much complicated and restrictive overhead: engine carts, generators, etc... I figured if we are going to be developing geo models to use with or on top of curation_concerns, we should be developing in that context.

I created a new rails app, installed the creation_concerns gem, and generated some initial 'geo works' following the patterns that Darren set out in #3. We now have a development framework that includes hyrda-jetty for testing as well as basic controllers and views to play with. When the models are more mature, we can easily pull them out into their own gem. 

#### Where's the good stuff at?

geo_concerns/app/models/
geo_concerns/app/models/concerns (where the real action is)
geo_concerns/app/models/concerns/basic_geo_metadata.rb (this is new)
geo_concerns/spec/ (you don't say)

Additionally, curation_concerns injects controllers, actors, and views for each model.

Is this a good way forward?
@jrgriffiniii @johnhuck

### CODE REVIEW

**Comment by drh-stanford** [comment]
I'm getting this error during `rake db:migrate` after a `bundle install`

```
NoMethodError: private method `include' called for ActiveFedora::Base:Class

/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activefedora-aggregation-0.4.0/lib/active_fedora/aggregation.rb:30:in `<module:Aggregation>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activefedora-aggregation-0.4.0/lib/active_fedora/aggregation.rb:8:in `<module:ActiveFedora>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activefedora-aggregation-0.4.0/lib/active_fedora/aggregation.rb:7:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/hydra-works-0.1.0/lib/hydra/works.rb:2:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/hydra-collections-6.0.0.alpha/lib/hydra-collections.rb:2:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/bundler/gems/curation_concerns-42738a70fa6a/lib/curation_concerns/engine.rb:4:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/bundler/gems/curation_concerns-42738a70fa6a/lib/curation_concerns.rb:2:in `<top (required)>'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:76:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:72:in `each'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:72:in `block in require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:61:in `each'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler/runtime.rb:61:in `require'
/Users/drh/.rvm/gems/ruby-2.0.0-p451/gems/bundler-1.10.6/lib/bundler.rb:134:in `require'
/Users/drh/workspace/pcdm-geo-models/geo_concerns/config/application.rb:7:in `<top (required)>'
/Users/drh/workspace/pcdm-geo-models/geo_concerns/Rakefile:4:in `require'
/Users/drh/workspace/pcdm-geo-models/geo_concerns/Rakefile:4:in `<top (required)>'

```

**Comment by drh-stanford** [comment]
The workaround to the error is to use Ruby 2.2.x.

**Comment by eliotjordan** [comment]
@drh-stanford I was wondering if the ruby version was the issue. Let me know if you have any other issues. CurationConcerns was just released on rubygems, so I'm going to test that now.

File: geo_concerns/app/models/concerns/basic_geo_metadata.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    property :georss_box, predicate: ::RDF::URI.new("http://www.georss.org/georss/box"), multiple: false do |index|
+      index.as :stored_searchable
+    end
+    
+    # Example values
+    #   urn:ogc:def:crs:EPSG::3163
+    #   urn:ogc:def:crs,crs:EPSG::4269,crs:EPSG::5713
+    #   http://www.opengis.net/def/crs/epsg/0/3163 (maybe? referred to in WFS query document)
+    # See http://portal.opengeospatial.org/files/?artifact_id=30575
+    property :crs, predicate: ::RDF::URI.new("http://www.opengis.net/def/dataType/OGC/1.1/crsURI"), multiple: false do |index|
```
Comment (drh-stanford) [inline geo_concerns/app/models/concerns/basic_geo_metadata.rb]: This should be at the Vector/Raster level, not basic geo

File: geo_concerns/app/models/image.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Base class for Images that are georeferenced with a bbox
+class Image < ActiveFedora::Base
+  include ::GeoConcerns::ImageBehavior
+  validates_presence_of :title,  message: 'Your work must have a title.'
+  validates_presence_of :georss_box,  message: 'Your work must have a bbox.'
```
Comment (drh-stanford) [inline geo_concerns/app/models/image.rb]: If we want to enforce bbox's at the Image level, then I'd say to enforce it at the basic geo metadata level. 

Same with title.

Both would keep the code DRY for these `validate_presence_of` calls.

**Comment by eliotjordan** [comment]
The curation_concerns gem from rubygems is not ready for primetime, it seems. I'll update the Gemfile when it is.

**Comment by drh-stanford** [comment]
I added PR #5 to address the comment about georectification behaviors being at the Raster/Vector level rather than Image

File: geo_concerns/app/models/image.rb (hunk 2 above)
Comment (eliotjordan) [inline geo_concerns/app/models/image.rb]: Good call. We should determine what properties should be enforced.

1. title
2. bbox
3. ??

Comment (drh-stanford) [inline geo_concerns/app/models/image.rb]: For Image, it'd be width and height, right? Then all geo classes would be title, bbox. For Vector/Raster we'd also include crs. I'm assuming that the `validates_presence_of` is for the properties(?)

**Comment by eliotjordan** [comment]
@drh-stanford Merge at your leisure. :palm_tree:

**Pull request merged**